src/Foundation/Connect/code/Managers/Order/OrderManagerV2.cs

```diff
@@ -0,0 +1,118 @@
+//    limitations under the License.
+
+namespace Wooli.Foundation.Connect.Managers.Order
+{
+    using System;
+    using System.Linq;
+
+    using Base.Models.Logging;
+    using Base.Services.Logging;
+
+    using Providers.Contracts;
+
+    using Sitecore.Commerce.Entities.Carts;
+    using Sitecore.Commerce.Services.Orders;
+    using Sitecore.Diagnostics;
+
+    public class OrderManagerV2 : BaseManager, IOrderManagerV2
+    {
+        private readonly OrderServiceProvider orderServiceProvider;
+
+        public OrderManagerV2(ILogService<CommonLog> logService, IConnectServiceProvider connectServiceProvider) : base(
+            logService)
+        {
+            Assert.ArgumentNotNull(connectServiceProvider, nameof(connectServiceProvider));
+
+            this.orderServiceProvider = connectServiceProvider.GetOrderServiceProvider();
+        }
+
+        public GetVisitorOrderResult GetOrder(string orderId, string customerId, string shopName)
+        {
+            Assert.ArgumentNotNullOrEmpty(orderId, nameof(orderId));
+            Assert.ArgumentNotNullOrEmpty(customerId, nameof(customerId));
+            Assert.ArgumentNotNullOrEmpty(shopName, nameof(shopName));
+
+            return this.Execute(
+                new GetVisitorOrderRequest(orderId, customerId, shopName),
+                this.orderServiceProvider.GetVisitorOrder);
+        }
+
+        public GetVisitorOrdersResult GetOrdersHeaders(string customerId, string shopName)
+        {
+            Assert.ArgumentNotNullOrEmpty(customerId, nameof(customerId));
+            Assert.ArgumentNotNullOrEmpty(shopName, nameof(shopName));
+
+            return this.Execute(
+                new GetVisitorOrdersRequest(customerId, shopName),
+                this.orderServiceProvider.GetVisitorOrders);
+        }
+
+        public SubmitVisitorOrderResult SubmitVisitorOrder(Cart cart)
+        {
+            Assert.ArgumentNotNull(cart, nameof(cart));
+
+            try
+            {
+                return this.Execute(
+                    new SubmitVisitorOrderRequest(cart),
+                    this.orderServiceProvider.SubmitVisitorOrder);
+            }
+            catch (Exception ex)
+            {
+                this.LogService.Error(ex.Message);
+
+                // ToDo: fix issues during order submitting: down there: temp implementation of the submit result
+                if (this.TryResolveSubmittedOrder(cart, out var managerResponse))
+                {
+                    return managerResponse;
+                }
+
+                throw;
+            }
+        }
+
+        private bool TryResolveSubmittedOrder(CartBase cart, out SubmitVisitorOrderResult managerResponse)
+        {
+            var getVisitorOrdersResult = this.GetOrdersHeaders(cart.CustomerId, cart.ShopName);
+
+            if (getVisitorOrdersResult.Success)
+            {
+                // Getting the latest order
```

Comment:
This comment is excessive. 
Please remove it in your next request OR if you think that it would be better for the next person to know that these 3 lines are `Getting the latest order`, create a private method with the descriptive name.